mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java

```diff
@@ -242,6 +242,11 @@ public Builder withInternalDate(Date date) {
                 return this;
             }
 
+            public Builder withInternalDate(Optional<Date> date) {
```

Comment:
can we put this commit in another PR please?

Reply:
No it is directly related.